File: src/libs/ThirdParty.VBSuppliers/Properties/YouTravel/YouTravelSupport.vb
Imports ThirdParty.Abstractions.Models

Public Class YouTravelSupport

    Public Shared Function GetDistinctDestinations(ByVal oResorts As List(Of ResortSplit)) As Generic.List(Of String)

        Dim oReturn As New Generic.List(Of String)
        Dim sDestination As String

        For Each oResort As ResortSplit In oResorts

            sDestination = oResort.ResortCode.Split("_"c)(0)
            If Not oReturn.Contains(sDestination) Then
                oReturn.Add(sDestination)
            End If

        Next

        Return oReturn
    End Function

    Public Shared Function FormatDate(ByVal dDate As Date) As String
        Return dDate.ToString("dd/MM/yyyy")
    End Function

End Class

File: src/libs/iVectorOne.Suppliers/YouTravel/YouTravelSupport.cs
namespace iVectorOne.Suppliers.YouTravel
{
    using System;
    using System.Collections.Generic;
    using iVectorOne.Models;
    using iVectorOne.Models.Property;

    public class YouTravelSupport
    {

        public static List<string> GetDistinctDestinations(List<ResortSplit> oResorts)
        {

            var oReturn = new List<string>();
            string sDestination;

            foreach (ResortSplit oResort in oResorts)
            {

                sDestination = oResort.ResortCode.Split('_')[0];
                if (!oReturn.Contains(sDestination))
                {
                    oReturn.Add(sDestination);
                }

            }

            return oReturn;
        }

        public static string FormatDate(DateTime dDate)
        {
            return dDate.ToString("dd/MM/yyyy");
        }

    }
}